experiments/prompt-injection-boundary-tags/rounds/round2b/harness/run_experiment.py

```
@@ -672,15 +864,28 @@ def run_experiment():
     print(f"  Combined results: {combined_csv}")
     print(f"{'='*70}")
 
+    report_path = budget.write_report(
+        completed_trials=budget_completed_trials,
+        stop_reason=budget_stop_reason,
+    )
+    if report_path:
+        print(f"Wrote budget report: {report_path}")
+
     # Final summary table
     print(f"\n📊 SUMMARY BY MODEL × CONDITION")
```

Ruff (0.15.0)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)